src/main/groovy/wooga/gradle/snyk/tasks/SnykTask.groovy

```diff
@@ -47,47 +49,60 @@ abstract class SnykTask extends DefaultTask implements SnykActionSpec, SnykTaskS
         logFile.set(value)
     }
 
-    private final ListProperty<String> additionalArguments = project.objects.listProperty(String)
-
-    @Override
-    void setAdditionalArguments(Iterable<String> value) {
-        additionalArguments.set(value)
-    }
-
-    @Override
-    void setAdditionalArguments(Provider<? extends Iterable<String>> value) {
-        additionalArguments.set(value)
-    }
-
-    @Override
-    void argument(String argument) {
-        additionalArguments.add(argument)
-    }
+    abstract void addMainOptions(List<String> args)
 
-    @Override
-    void arguments(String... arguments) {
-        additionalArguments.addAll(arguments)
-    }
-
-    @Override
-    void arguments(Iterable<String> arguments) {
-        additionalArguments.addAll(arguments)
+    SnykTask() {
+        arguments.set(project.provider({ composeArguments() }))
     }
 
     @TaskAction
     protected void exec() {
-        def arguments = arguments.get()
+
+        def _arguments = getAllArguments()
+        def _workingDir = workingDirectory.getOrNull()
+        def _executable
+        if (snykPath.isPresent()) {
+            _executable = snykPath.get().asFile.path
+        } else {
+            logger.warn("No path for the snyk executable was set; will invoke ${executableName.get()} in the default path ")
+            _executable = executableName.get()
+        }
+
+        ExecResult execResult = project.exec(new Action<ExecSpec>() {
+            @Override
+            void execute(ExecSpec exec) {
+
+                logger.info("Invoking snyk executable (${_executable}) with ${_arguments}")
+
+                exec.with {
+                    executable _executable
+                    args = _arguments
+                    ignoreExitValue = true
+                    // Optional working directory
+                    if (_workingDir != null) {
+                        workingDir(_workingDir)
+                    }
+                }
+            }
+        })
+        handleExitCode(execResult.exitValue)
```

Comment:
The whole logfile handling is missing here. But we can add this in the next patch

Reply:
Understood.